control-plane/src/modules/jwt.ts

```diff
@@ -8,28 +8,59 @@ export class AuthenticationError extends Error {
   }
 }
 
-if (!process.env.JWKS_URL) {
-  throw new Error("JWKS_URL must be set");
+if (!process.env.JWKS_URL && !process.env.MANAGEMENT_SECRET) {
+  throw new Error("No JWKS_URL or MANAGEMENT_SECRET in env. One is required.");
 }
 
-const client = jwksClient({
-  jwksUri: process.env.JWKS_URL,
-});
+if (process.env.MANAGEMENT_SECRET) {
+  const hasPrefix = process.env.MANAGEMENT_SECRET.startsWith("sk_management_");
+  const hasLength = process.env.MANAGEMENT_SECRET.length > 64;
+
+  if (!hasPrefix) {
+    throw new Error("MANAGEMENT_SECRET must start with sk_management_");
+  }
+
+  if (!hasLength) {
+    throw new Error("MANAGEMENT_SECRET must be longer than 64 characters");
+  }
+}
+
+const client = process.env.JWKS_URL
+  ? jwksClient({
+      jwksUri: process.env.JWKS_URL,
+    })
+  : null;
 
 const getKey: GetPublicKeyOrSecret = (header, callback) => {
-  return client.getSigningKey(header.kid, function (err, key) {
+  return client?.getSigningKey(header.kid, function (err, key) {
```

Comment:
We could probably throw here If `client` is `undefined`?

Reply:
Thanks. Done!